pkg/collector/client.go

```diff
@@ -11,28 +11,59 @@ import (
 type clientCollector struct {
 	omadaClientDownloadActivityBytes *prometheus.Desc
 	omadaClientSignalDbm             *prometheus.Desc
+	omadaClientRssiDbm				 *prometheus.Desc
+	omadaClientTrafficDown			 *prometheus.Desc
+	omadaClientTrafficUp			 *prometheus.Desc
+	omadaClientTxRate				 *prometheus.Desc
+	omadaClientRxRate				 *prometheus.Desc
+
 	omadaClientConnectedTotal        *prometheus.Desc
+	omadaClientWirelessTotal         *prometheus.Desc
+	omadaClientWireless2g            *prometheus.Desc
+	omadaClientWireless5g            *prometheus.Desc
+	omadaClientWireless6g            *prometheus.Desc
+	omadaClientWiredTotal         	 *prometheus.Desc
 	client                           *api.Client
 }
 
 func (c *clientCollector) Describe(ch chan<- *prometheus.Desc) {
 	ch <- c.omadaClientDownloadActivityBytes
 	ch <- c.omadaClientSignalDbm
+	ch <- c.omadaClientRssiDbm		
+	ch <- c.omadaClientTrafficDown	
+	ch <- c.omadaClientTrafficUp	
+	ch <- c.omadaClientTxRate		
+	ch <- c.omadaClientRxRate	
 	ch <- c.omadaClientConnectedTotal
+	ch <- c.omadaClientWirelessTotal
+	ch <- c.omadaClientWireless2g
+	ch <- c.omadaClientWireless5g
+	ch <- c.omadaClientWireless6g
+	ch <- c.omadaClientWiredTotal
 }
 
 func (c *clientCollector) Collect(ch chan<- prometheus.Metric) {
 	client := c.client
 	config := c.client.Config
 
 	site := config.Site
-	clients, err := client.GetClients()
+	clients, stats, err := client.GetClients()
 	if err != nil {
 		log.Error().Err(err).Msg("Failed to get clients")
 		return
 	}
 
-	ch <- prometheus.MustNewConstMetric(c.omadaClientConnectedTotal, prometheus.GaugeValue, float64(len(clients)),
+	ch <- prometheus.MustNewConstMetric(c.omadaClientConnectedTotal, prometheus.GaugeValue, stats.Total,
```

Comment:
Instead of creating new metrics for `omada_client_wireless_2g`, `omada_client_wireless_5g`, `omada_client_wired_total` etc. we should instead follow the Prometheus model for metric dimensions and a label to this metric.

We can add labels for `type="wired"` and `type="wireless"` and also, we already have the `wifi_mode` label which tells us what protocol the wireless connection is using. It's currently just a weird number that doesn't mean anything, but, [I managed to find a link to the API documentation](https://community.tp-link.com/en/business/forum/topic/590430) and it documents what protocol each number represents:
```
0: 11a  1: 11b  2: 11g  3: 11na  4: 11ng  5: 11ac  6: 11axa  7: 11axg
```

As we're already looping through all the available clients, we should count each client type in a variable and then apply the values with labels at the end.

Reply:
I can't comment on the line as it hasn't been changed, [but this is what will need mapping from a number to the actual protocol](https://github.com/charlie-haley/omada_exporter/blob/main/pkg/collector/client.go#L42)

Reply:
I agree that adding labels to clients could be a good move and the mapping that you included definitely helps.
The best solution is probably to not even have a "*total" metrics, but instead to just count another metric and group by / filter by the `type` / `wifi_mode` labels.

At the same time, the stats are reported by Omada, align with the existence of the existing `omada_client_connected_total`  metric and to be frank was just simple :)

Will it make sense to add the labels to `omada_client_connected_total` so one can just filter on the labels if needed?

Reply:
See recent changes